src/plugins/mer/mersettings.cpp

```diff
@@ -209,17 +277,38 @@ void MerSettings::read()
             true).toBool();
     m_qmlLiveBenchLocation = settings->value(QLatin1String(QML_LIVE_BENCH_LOCATION_KEY)).toString();
     m_askBeforeStartingVmEnabled = settings->value(QLatin1String(ASK_BEFORE_STARTING_VM), true).toBool();
-    m_askBeforeClosingVmEnabled = settings->value(QLatin1String(ASK_BEFORE_STARTING_VM), true).toBool();
+    m_askBeforeClosingVmEnabled = settings->value(QLatin1String(ASK_BEFORE_STARTING_VM), true).toBool();    
 
     settings->endGroup();
 
+    settings->beginGroup(QLatin1String(SETTINGS_EMULATOR_CATEGORY));
+    foreach (const QString &name, settings->childKeys()) {
+        const auto map = settings->value(name).value<QVariantMap>();
+        MerEmulatorDeviceModel emulator;
+        emulator.fromMap(map);
+        m_customEmulatorDeviceModels.insert(name, emulator);
+    }
+    settings->endGroup();
+
     m_environmentFilterFromEnvironment =
         QProcessEnvironment::systemEnvironment().value(Constants::SAILFISH_OS_SDK_ENVIRONMENT_FILTER);
 }
 
-void MerSettings::save()
+void MerSettings::saveCustomEmulatorDeviceModels()
```

Comment:
Inlinize to save() as you did for read()?

Reply:
The reason is to create and delete emulator profile in settings file as soon as user press Add/Remove button.

Reply:
You shouln't commit any changes until the Accept/OK button is used.

Reply:
The issue resolved.